### PR TITLE
Integrate item favoriting logic and ui

### PR DESCRIPTION
**Pull request by amjiao**
## Overview

Completed item favoriting screen

## Changes Made

Item favoriting screen, new ItemFavoritesFilterRow, added some filtering logic to FavoritesViewModel  

## Test Coverage

Clicked all of the buttons and they worked!

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/FavoritesToggle.kt
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/FavoritesToggle.kt:39]: Hmm okay this is a bit sus, the existence of separate `Composable`s for `ActiveToggle` and `InactiveToggle`. The two functions are VERY similar, in that there's a good amount of code reuse with perhaps a few lines that are only slightly different.

Let's make this a `FavoritesToggle` function that takes in an `active: Boolean` parameter instead. If it's active, the correct colors will be set appropriately--if it's inactive, same thing.

That way we cut down on code. Makes it more readable and arguably easier to interface with--all you need to do is pass in `active = true` to the toggle that's supposed to be true.

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/FavoritesToggle.kt
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/FavoritesToggle.kt:88]: Pro tip! If you want your preview functions to actually DO something (aka showcase some behavior) you can add a `var active by remember { mutableStateOf(true) }` and cause the `{}` anonymous functions to instead set active to true / false, and then finally (with my other suggestion) pass this `active` value into the components so that you can actually see the toggle working in your preview. If you enter `Interactive Mode` in the preview, you can actually tap them and see the toggle change over. 

Great way to test for intended behavior!

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/general/FilterRow.kt
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/components/general/FilterRow.kt:228]: The way this whole file is structured is so sus. We should really try to make these `items` instead of `item` if possible. Zach's PR touches on this, but that needs to be merged first before we tackle this.

Plz add a TODO to this (and the other sus filter row composables here) to make them more concise :-)

File: app/src/main/java/com/cornellappdev/android/eatery/ui/navigation/MainTabbedNavigation.kt
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/navigation/MainTabbedNavigation.kt:371]: Formatting is sus >:(

I think Zach fixed this instance on his PR but plz make sure to format code with ctrl alt L :-)

File: app/src/main/java/com/cornellappdev/android/eatery/ui/screens/FavoritesScreen.kt
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/screens/FavoritesScreen.kt:200]: Okay yeah so once you get to the refactor of this, you should then be able to make this NOT AT ALL need an if/else statement

File: app/src/main/java/com/cornellappdev/android/eatery/ui/screens/FavoritesScreen.kt
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/screens/FavoritesScreen.kt:311]: This file's getting a bit long LOL, can you pull this out into a separate file under the `components` directory somewhere?

File: app/src/main/java/com/cornellappdev/android/eatery/ui/screens/FavoritesScreen.kt
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/screens/FavoritesScreen.kt:382]: ...along with this :-)

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/FavoritesToggle.kt
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/FavoritesToggle.kt:12]: Android studio's telling me there are an asston of unused imports here, can u run `Optimize Imports` here to remove them? 
image

File: app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/ToggleViewModel.kt
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/ToggleViewModel.kt:22]: This seems a little unnecessary? All this is doing is basically communicating the toggle state of one boolean right?

I do believe that Zach took this out in his cleanup though. At least, I know he took it out of `FavoritesScreen.kt`. If after his merge goes in this file is not being used anymore, let's delete it.

**Review by thisjustin123 — CHANGES_REQUESTED** [review]
Great stuff!! The UI looks great and generally the flow of info with VMs is looking great, impressive stuff picking all that up so fast :D

I definitely have some points to consider, so plz take a look :-)

and also let's hold off on merging this PR until Zach merges his first INTO this one.

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/FavoritesToggle.kt
Comment (amjiao) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/FavoritesToggle.kt:12]: Done!

File: app/src/main/java/com/cornellappdev/android/eatery/ui/navigation/MainTabbedNavigation.kt
Comment (amjiao) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/navigation/MainTabbedNavigation.kt:371]: done!

File: app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/ToggleViewModel.kt
Comment (amjiao) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/ToggleViewModel.kt:22]: Makes sense, I moved the toggle functions to the FavoritesViewModel and deleted ToggleViewModel

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/general/FilterRow.kt
Comment (amjiao) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/components/general/FilterRow.kt:228]: done!

**Review by thisjustin123 — APPROVED** [review]
Really great stuff, thanks for responding to my comments so fast and for asking great questions! Just make sure to re-request my review next time :D

**Pull request merged**
